api/blueprints/drbCitation.py

```diff
@@ -59,6 +60,777 @@ def citationFetch(uuid):
+            editorFirstName = listEditorName[1]
+            authName = citationWork.authors[0]['name']
+            citationResponse = f"{authName}. {citationWork.title}, \
+                edited by {editorFirstName} {editorLastName}, {citationWork.editions[0].publication_place}, \
+                {citationWork.editions[0].publication_date}."
+
+            return ' '.join(citationResponse.split())
+
+    #Translated works/works prepared by editors during or after 1900
+    if citationWork.editions[0].publication_date >= date(1900, 1, 1) and \
+        citationWork.contributors:
+
+        i = 0
+        transName = ''
+        editorName = ''
+
+        while i < len(citationWork.contributors):
+            if citationWork.contributors[i]['roles'] == 'translator':
+                transName = citationWork.contributors[i]['roles']
+            if citationWork.contributors[i]['roles'] == 'editor':
+                editorName = citationWork.contributors[i]['roles']
+            i += 1
+        
+        if transName and editorName:
+            listTransName = transName.split(', ')
+            transLastName = listTransName[0]
+            transFirstName = listTransName[1]
+            listEditorName = editorName.split(', ')
+            editorLastName = listEditorName[0]
+            editorFirstName = listEditorName[1]
+            authName = citationWork.authors[0]['name']
+
+            citationResponse = f"{authName}. {citationWork.title}. \
+                                translated by {transFirstName} {transLastName}, \
+                                edited by {editorFirstName} {editorLastName}, \
+                                {citationWork.editions[0].publishers[0]['name']}, \
+                                {citationWork.editions[0].publication_date}."
+
+            return ' '.join(citationResponse.split())
+                    
+        if transName:
+            listTransName = transName.split(', ')
+            transLastName = listTransName[0]
+            transFirstName = listTransName[1]
+            citationResponse = f"{authName}. {citationWork.title}. \
+                                translated by {transFirstName} {transLastName}, \
+                                {citationWork.editions[0].publishers[0]['name']}, \
+                                {citationWork.editions[0].publication_date}."
+
+            return ' '.join(citationResponse.split())
+
+        if editorName:
+            listEditorName = editorName.split(', ')
+            editorLastName = listEditorName[0]
+            editorFirstName = listEditorName[1]
+            citationResponse = f"{authName}. {citationWork.title}, \
+                                edited by {editorFirstName} {editorLastName}, \
+                                {citationWork.editions[0].publishers[0]['name']}, \
+                                {citationWork.editions[0].publication_date}."
+
+            return ' '.join(citationResponse.split())
```

Comment:
The issue mentioned in the PR comment stems from here, I think. There is no case for handling a book with no translator or editor, which results in a return of `None`.

I think this issue is one that will get handled with a refactor so that we can implement a base case for these conditions that can be shared across the formatting rules!